### ISSUE TITLE
Image scaling

### ISSUE THREAD

**Issue by d0t101101** [igorgladkov/rotini#30]
The new widget size control is helpful for most things, but I've found truncation occurs when using an image/map/webview widget.  For instance:

Location - would you please make the map(s) consistently render render full page if no 'size' is specified, and/or maybe the 'full' option described below?  Today, not specifying the size makes it large but not full height/width depending on orientation.  Played with various size options and could not get it to work as expected:

```
//Location Map
Text label="Location" {
  Frame label="Location {widget:map-advanced}" {
  Text label="{item:target} [37.935261,-76.048225]"
  Text label="{item:zoom} [12]"
}
```

Image - I use an 'image' to display my webcam and floorplan layout (png file); and for each the left/right edges of the image are cut off.  The truncation becomes WORSE in landscape orientation when compared to portrait (as Rotini tries to stuff all content into a widget, two column, side-by-side).

```
 //PTZ1
  Frame label="{widget:image}" {
    Image url="http://IP:PORT/CGIProxy.fcgi?cmd=snapPicture2&usr=USERNAME&pwd=PASSWORD" refresh=1000
}
```

RRD Webview - For a temp workaround until charts are hopefully natively implemented, I've played with the size option to add RRD charts to a dedicated 'charts' page via the web widget:

```
//RRD Charts Page
 Text label="Charts" {
  //Indoor Temp RRD
   Frame label="{widget:web,size:2x1}" {
   Webview url="/chart?groups=HVAC_Chart_Temp&period=d" 
 }
 //Indoor Humidity RRD
 Frame label="{widget:web,size:2x1}" {
   Webview url="/chart?groups=HVAC_Chart_Humidity&period=d" 
 }
```

   }

This works for the most part, but the problem is these always truncate the bottom to make them fit into the widget dimensions, so users can't see the lower time axis of the charts...

I'd suggest adding a 'size:full' option which keeps the widget stretched to the full pixel-perfect length/width of the source image and display, regardless of orientation.

**Comment by igorgladkov** [comment]
First, there is a size: "fill" option already that I think will do what you asking for. It is not a default for any widget, since most of the widgets are sharing screen with other widgets, while "fill" only makes sense when there is only one widget on a page.

Second, image truncation does occur depending on the aspect ratio of the image. I'm working on possible solutions (I will also check on web page cut offs).

**Comment by d0t101101** [comment]
'size:fill' was exactly what I was after, thanks!
Looking forward to what you come up with for the aspect ratio issues.

Much appreciated,
.

**Comment by igorgladkov** [comment]
Fixed in 0.2.20

**Comment by d0t101101** [comment]
This definitely fixed image widgets, but the same issue seems to apply to webview widgets.  Albeit, these can be quite a bit more complicated!

Thnx,
.

**Comment by igorgladkov** [comment]
Webview does not do any special scaling on the pages, so it page is mobile compatible (responsive) then it should work as is. Are you talking about some page with yours or outside of your control?

**Comment by d0t101101** [comment]
Ah, found an error in my config which explains this.  I was trying to load RRD charts initially using webview, and switched to the image widget after this recent update.  The problem was I changed the item WITHIN the frame but missed the widget itself (the frame label), hence images were being cropped heavily as they were still in webview mode.  This resulted in a mix of webview widgets with images beneath...

Switching everything over from webview to image fixed the issue, and my RRD charts are usable!  While the dimensions are not perfect, this works fix certainly for me :)

Thanks again, Igor!

**Comment by igorgladkov** [comment]
@d0t101101 Can you post a screenshot of how RRD chart looks in image widget?

**Comment by d0t101101** [comment]
Sure, you are go.  I played with chart:scale options in openhab.cfg, which doesn't appear to have an effect in Rotini.  First screenshot is using images defined like so:

```
//Indoor Temp RRD
Frame label="{widget:image,size:2x1}" {
  Image url="/chart?groups=HVAC_Chart_Temp&period=d" refresh=10000
}
```

![screenshot1](https://cloud.githubusercontent.com/assets/4561321/12379018/8d73ff6e-bd1e-11e5-956f-dd4547c97f87.png)

Second screenshot I switched to fill.  This gives better results in landscape mode, but in portrait there is the large black borders:

```
//Indoor Temp RRD
Frame label="{widget:image,size:fill}" {
  Image url="/chart?groups=HVAC_Chart_Temp&period=d" refresh=10000
}
```

![screenshot2](https://cloud.githubusercontent.com/assets/4561321/12379019/927498f2-bd1e-11e5-9d4d-8248470c3fe8.png)

All the best,
.
